### PR TITLE
Set output blobs precision for IE tests

### PR DESCRIPTION
**Pull request by mbencer**
Tikcet: 46770

Context:
CPU plugin performs output precision implicit conversion (to `float` or `int32`).

For some cases the solution is calling `setPrecision` directly on output blob (based on nG Results element type).

### CODE REVIEW

File: ngraph/test/util/engine/ie_engines.cpp [hunk 1]
```diff
@@ -123,6 +123,41 @@ namespace
...
+namespace
+{
+    InferenceEngine::Precision ng_type_to_precission(const element::Type& target_type)
+    {
+#if defined(__GNUC__) && !(__GNUC__ == 4 && __GNUC_MINOR__ == 8)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic error "-Wswitch"
+#pragma GCC diagnostic error "-Wswitch-enum"
+#endif
+        switch (target_type)
```
Comment (tsocha) [inline ngraph/test/util/engine/ie_engines.cpp:135]: It's surprising that there was no such switch before.

File: ngraph/test/util/engine/ie_engines.cpp [hunk 2]
```diff
@@ -172,6 +214,32 @@ testing::AssertionResult test::IE_Engine::compare_results(const size_t tolerance
...
 }
 
+std::string test::IE_Engine::get_output_name(const std::shared_ptr<op::v0::Result>& ng_result)
+{
+    if (m_function->get_results().size() == 1)
+    {
+        // ng_result argument is ignored
+        return m_network_outputs.begin()->first;
+    }
+    else
```
Comment (tsocha) [inline ngraph/test/util/engine/ie_engines.cpp:224]: Do test_case support networks with multiple outputs?
I know there were some problems with it.

**Review by tsocha — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: ngraph/test/util/engine/ie_engines.cpp (hunk 1 above)
Comment (mbencer) [reply ngraph/test/util/engine/ie_engines.cpp:135]: That's right: https://github.com/openvinotoolkit/openvino/blob/ab66eab6528b2ec99cc992f07ffd3dcb2950b8bd/inference-engine/src/plugin_api/ie_ngraph_utils.hpp#L93, but it is not part of public IE API

File: ngraph/test/util/engine/ie_engines.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/test/util/engine/ie_engines.cpp:224]: It works for multiple outputs. The logic of calculating output name is the same as before. I moved it only to helper method.
For single output `m_network_outputs.begin()->first;` is always returned (result argument is not used).
For multioutput case the name is based on previous node name (plus dot).

**Review by tomdol — APPROVED** [review]

**Pull request merged**